src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -569,6 +567,80 @@ private void OutputResultsHelper(IEnumerable<CommandInfo> results)
 #endif
         }
 
+        /// <summary>
+        /// Creates the syntax output based on if the command is an alias, script, application or command.
+        /// </summary>
+        /// <param name="command">
+        /// CommandInfo object containing the syntax to be output.
+        /// </param>
+        /// <returns>
+        /// Syntax string cast as a PSObject for outputting.
+        /// </returns>
+        private PSObject GetSyntaxObject(CommandInfo command)
+        {
+            PSObject syntax = PSObject.AsPSObject(command.Syntax);
+
+            if (this.Name != null && !Array.Exists(this.Name, name => name.Equals(command.Name, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Should add a comment here explaining this line like you explained it in the resolved conversation

Reply:
Added.